addons/http_routing/models/ir_http.py

```diff
@@ -217,26 +217,24 @@ def is_multilang_url(local_url, lang_url_codes=None):
 
     url = local_url.partition('#')[0].split('?')
     path = url[0]
+
+    # Consider /static/ and /web/ files as non-multilang
+    if '/static/' in path or path.startswith('/web/'):
+        return False
+
     query_string = url[1] if len(url) > 1 else None
     router = request.httprequest.app.get_db_router(request.db).bind('')
 
     def is_multilang_func(func):
         return (func and func.routing.get('website', False) and
                 func.routing.get('multilang', func.routing['type'] == 'http'))
+
     # Try to match an endpoint in werkzeug's routing table
     try:
-        func = router.match(path, method='POST', query_args=query_string)[0]
-        return is_multilang_func(func)
-    except werkzeug.exceptions.MethodNotAllowed:
-        func = router.match(path, method='GET', query_args=query_string)[0]
-        return is_multilang_func(func)
-    except werkzeug.exceptions.NotFound:
-        # Consider /static/ files as non-multilang
-        static_index = path.find('/static/', 1)
-        if static_index != -1 and static_index == path.find('/', 1):
-            return False
-        return True
-    except Exception:
+        new_url, func = request.env['ir.http']._url_rewrite_kargs(path, query_args=query_string)
+        return is_multilang_func(func) if func else True  # /page/xxx has not endpoint/func but multilang
```

Comment:
`is_multilang_func()` is now used only once, so you could inline it here perhaps?

```suggestion
        # /page/xxx has no endpoint/func but multilang
        return (not func or (
            func.routing.get('website') and
            func.routing.get('multilang', func.routing['type'] == 'http')))
```